                                  # Heart_Disease_Prediction&Algorithm_Comparison-SVM-KNN                      

### heart disease prediction using SVM and KNN algorithms. In this project, we will use the UCI Heart Disease dataset to build and compare the performance of Support Vector Machine (SVM) and K-Nearest Neighbors (KNN) algorithms for predicting heart disease.

# Import The Data

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('https://raw.githubusercontent.com/itsluckysharma01/Datasets/refs/heads/main/heart_disease_uci.csv')

In [3]:
df

id  age     sex        dataset               cp  trestbps   chol    fbs  \
0      1   63    Male      Cleveland   typical angina     145.0  233.0   True   
1      2   67    Male      Cleveland     asymptomatic     160.0  286.0  False   
2      3   67    Male      Cleveland     asymptomatic     120.0  229.0  False   
3      4   37    Male      Cleveland      non-anginal     130.0  250.0  False   
4      5   41  Female      Cleveland  atypical angina     130.0  204.0  False   
..   ...  ...     ...            ...              ...       ...    ...    ...   
915  916   54  Female  VA Long Beach     asymptomatic     127.0  333.0   True   
916  917   62    Male  VA Long Beach   typical angina       NaN  139.0  False   
917  918   55    Male  VA Long Beach     asymptomatic     122.0  223.0   True   
918  919   58    Male  VA Long Beach     asymptomatic       NaN  385.0   True   
919  920   62    Male  VA Long Beach  atypical angina     120.0  254.0  False   

              restecg  thalch  exang  oldpeak        slope   ca  \
0      lv hypertrophy   150.0  False      2.3  downsloping  0.0   
1      lv hypertrophy   108.0   True      1.5         flat  3.0   
2      lv hypertrophy   129.0   True      2.6         flat  2.0   
3              normal   187.0  False      3.5  downsloping  0.0   
4      lv hypertrophy   172.0  False      1.4    upsloping  0.0   
..                ...     ...    ...      ...          ...  ...   
915  st-t abnormality   154.0  False      0.0          NaN  NaN   
916  st-t abnormality     NaN    NaN      NaN          NaN  NaN   
917  st-t abnormality   100.0  False      0.0          NaN  NaN   
918    lv hypertrophy     NaN    NaN      NaN          NaN  NaN   
919    lv hypertrophy    93.0   True      0.0          NaN  NaN   

                  thal  num  
0         fixed defect    0  
1               normal    2  
2    reversable defect    1  
3               normal    0  
4               normal    0  
..                 ...  ...  
915                NaN    1  
916                NaN    0  
917       fixed defect    2  
918                NaN    0  
919                NaN    1  

[920 rows x 16 columns]

# Dataset Column Descriptions

## Feature Variables

| Column | Description |
|--------|-------------|
| **id** | Unique identifier for each patient record |
| **age** | Patient's age in years |
| **sex** | Patient's gender (Male/Female) |
| **dataset** | Source dataset (Cleveland in this case) |
| **cp** | Chest pain type |
| **trestbps** | Resting blood pressure (in mm Hg) |
| **chol** | Serum cholesterol level (in mg/dl) |
| **fbs** | Fasting blood sugar > 120 mg/dl (True/False) |
| **restecg** | Resting electrocardiographic results |
| **thalch** | Maximum heart rate achieved during exercise |
| **exang** | Exercise induced angina (True/False) |
| **oldpeak** | ST depression induced by exercise relative to rest |
| **slope** | Slope of the peak exercise ST segment |
| **ca** | Number of major vessels (0-3) colored by fluoroscopy |
| **thal** | Thalassemia type |

## Categorical Variable Details

### Chest Pain Type (cp):
- **typical angina**: Classic chest pain related to heart disease
- **atypical angina**: Chest pain not typical of heart disease
- **non-anginal**: Chest pain not related to heart disease
- **asymptomatic**: No chest pain symptoms

### Resting ECG Results (restecg):
- **normal**: Normal electrocardiogram
- **having ST-T wave abnormality**: Abnormal ST-T wave patterns
- **lv hypertrophy**: Left ventricular hypertrophy (enlarged heart muscle)

### ST Segment Slope (slope):
- **upsloping**: Upward sloping ST segment
- **flat**: Flat ST segment
- **downsloping**: Downward sloping ST segment

### Thalassemia Type (thal):
- **normal**: Normal thalassemia
- **fixed defect**: Fixed defect in thalassemia
- **reversable defect**: Reversible defect in thalassemia

## Target Variable

| Column | Description |
|--------|-------------|
| **num** | **Target variable** - Presence of heart disease |

**Target Values:**
- **0**: No heart disease
- **1-4**: Varying degrees of heart disease presence (1 = mild, 4 = severe)

---

**Note:** The `num` column is your target variable for prediction in this heart disease classification task.